weed/filer2/filerstore.go

```diff
@@ -129,6 +131,54 @@ func (fsw *FilerStoreWrapper) ListDirectoryEntries(ctx context.Context, dirPath
 	return entries, err
 }
 
+func (fsw *FilerStoreWrapper) ListDirectoryPrefixedEntries(ctx context.Context, dirPath util.FullPath, startFileName string, includeStartFile bool, limit int, prefix string) ([]*Entry, error) {
+	stats.FilerStoreCounter.WithLabelValues(fsw.ActualStore.GetName(), "list").Inc()
+	start := time.Now()
+	defer func() {
+		stats.FilerStoreHistogram.WithLabelValues(fsw.ActualStore.GetName(), "list").Observe(time.Since(start).Seconds())
+	}()
+	entries, err := fsw.ActualStore.ListDirectoryPrefixedEntries(ctx, dirPath, startFileName, includeStartFile, limit, prefix)
+	if err.Error() == UnsupportedListDirectoryPrefixedErr.Error() {
+		count := 0
+		notPrefixed, err := fsw.ActualStore.ListDirectoryEntries(ctx, dirPath, startFileName, includeStartFile, limit)
+		if err != nil {
+			return nil, err
+		}
+		if prefix == "" {
+			entries = notPrefixed
+		} else {
+			var lastFileName string
+			for count < limit {
+				for _, entry := range notPrefixed {
+					lastFileName = entry.Name()
+					if strings.HasPrefix(entry.Name(), prefix) {
+						count++
+						entries = append(entries, entry)
+					}
+				}
+				if count >= limit {
+					break
+				}
+
+				notPrefixed, err = fsw.ActualStore.ListDirectoryEntries(ctx, dirPath, lastFileName, includeStartFile, limit)
```

Comment:
the limit can be `limit - count`

Reply:
Its seemes limit is pagination limit and it makes no sense to reduce by count